xsdata/models/elements.py

```diff
@@ -429,39 +495,44 @@ def extends(self) -> Optional[str]:
 @dataclass
 class Extension(AnnotationBase):
     """
-    The extension element extends an existing simpleType or complexType
-    element.
-
-    Reference: https://www.w3schools.com/xml/el_extension.asp.
+    <extension
+      base = QName
+      id = ID
+      {any attributes with non-schema namespace . . .}>
+      Content: (annotation?, ((attribute | attributeGroup)*, anyAttribute?), assert*)
+    </extension>
     """
 
-    base: Optional[str] = attribute(default=None)
-    group: Optional[Group] = element(default=None)
-    all: Optional[All] = element(default=None)
-    choice: Optional[Choice] = element(default=None)
-    sequence: Optional[Sequence] = element(default=None)
-    any_attribute: Optional[AnyAttribute] = element(default=None)
-    attributes: ArrayList[Attribute] = element(default_factory=list, name="attribute")
-    attribute_groups: ArrayList[AttributeGroup] = element(
-        default_factory=list, name="attributeGroup"
-    )
-    assertions: ArrayList[Assertion] = element(default_factory=list, name="assert")
+    base: Optional[str] = attribute()
+    group: Optional[Group] = element()
+    all: Optional[All] = element()
+    choice: Optional[Choice] = element()
+    sequence: Optional[Sequence] = element()
+    any_attribute: Optional[AnyAttribute] = element()
+    attributes: Array[Attribute] = array_element(name="attribute")
+    attribute_groups: Array[AttributeGroup] = array_element(name="attributeGroup")
+    assertions: Array[Assertion] = array_element(name="assert")
 
     @property
     def extends(self) -> Optional[str]:
         return self.base
 
+    def get_restrictions(self) -> Dict[str, Any]:
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/tefra/xsdata/pullRequest?prid=5052528)